src/core/StelLocation.hpp

```diff
@@ -66,6 +66,13 @@ class StelLocation
 	//! @return azimuth counted from north or south as set in the StelApp preferences, in [0...360].
 	double getAzimuthForLocation(double longTarget, double latTarget) const;
 
+	//! longitude and latitude are private to force use of special getters. These return a location on the north pole if we are located on an "Observer".
+	//! By this we force useful view orientation.
+	float getLatitude()  const { if (planetName.endsWith(" Observer")) return 90.f; else return latitude;}
```

Comment:
[`StelLocation.cpp:228`](https://github.com/Stellarium/stellarium/blob/master/src/core/StelLocation.cpp#L228) shows a precedent of finding a `Planet` object. Same way can be used to check for the actual planet being an observer, rather than relying on its name.

Reply:
OK, this appears to be a serious mistake. After you fixed this I got a performance regression from 38 FPS to 1.2 FPS. I have 186 planets in the list, and the search function appears to be now called several thousand times per frame.

Reply:
I think the better fix would be to cache `PlanetP` in the `StelLocation` along with `planetName`.

Reply:
Ouch. I had feared that this was going to cost somewhat, but not that much. Just that the StelLocation has no pointer to its planet. 

Hmm, we can most likely use the Location's role for this! A small o can be used for "observer". I cannot continue today, though.

Reply:
> Just that the StelLocation has no pointer to its planet.

Why not introduce it? It would be updated whenever `planetName` changes.

Reply:
No need for another variable if `role` can do it. Hopefully tomorrow...